streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -92,7 +92,11 @@
     private final Map<TaskId, StandbyTask> standbyTasks;
     private final Map<TopicPartition, StreamTask> activeTasksByPartition;
     private final Map<TopicPartition, StandbyTask> standbyTasksByPartition;
+    private final Map<TopicPartition, StreamTask> suspendedTasksByPartition;
```

Comment:
We have a lot of task related Maps. I wonder if would could encapsulate them behind another Object? Blocks of code like lines 345 - 355 would then be a method on the new Object. Admittedly, i haven't completely thought it through...